keyboards/cablecardesigns/cypher/rev6/rev6.c

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 Cable Car Designs (@westfoxtrot)
+// SPDX-License-Identifier: GPL-2.0-or-later
+#include "rev6.h"
+
+// Indicator color definitions
+#ifndef HSV_CAPS
+    #define HSV_CAPS  96, 255, 120 // Define caps lock color (H, S, V)
+#endif
+
+#ifndef HSV_NLCK
+    #define HSV_NLCK  160, 255, 120 // Define num lock color (H, S, V)
+#endif
+
+#ifndef HSV_LYR
+    #define HSV_LYR  0, 255, 120 // Define layer color (H, S, V)
+#endif
+
+#ifndef HSV_BLACK
+    #define HSV_BLACK  0, 0, 0  // Define 'black' color, more like 'LED off' (H, S, V)
+#endif
+
+void eeconfig_init_kb(void) {  // EEPROM is getting reset!
+  rgblight_enable(); // Enable RGB by default
+  rgblight_sethsv(0, 255, 128);  // Set default HSV - red hue, full saturation, medium brightness
+  rgblight_mode(RGBLIGHT_MODE_RAINBOW_SWIRL + 2); // set to RGB_RAINBOW_SWIRL by default
+
+  eeconfig_update_kb(0);
+  eeconfig_init_user();
+}
+
+#ifdef RGBLIGHT_ENABLE
+
+__attribute__ ((weak))
```

Comment:
I'd be reluctant to put `weak` user-level implementations here - they're already defined as `weak` in core, and is very much dependent on the order that the files get linked together as far as which one takes precedence. You may find that in certain environments the implementations in this file are completely disregarded.

The usual method is to place them in the keymaps themselves, without the weak attribute. Whilst it does add code duplication, it removes the ambiguity and the reliance on the build tooling quirks.